File: Python/backend_flask_archi_feature/author/author_module.py
from author.db.author_inmemory_mapper import AuthorInMemoryMapper
from author.db.author_inmemory_repository import AuthorInMemoryRepository
from author.db.author_db import author_db
from author.dto.author_dto_mapper import AuthorDTOMapper
from author.dto.author_dto import author_dto
from author.author_service import AuthorService

_author_db_mapper = AuthorInMemoryMapper()
_author_repository = AuthorInMemoryRepository(database=author_db, mapper=_author_db_mapper)

_author_service = AuthorService(repository=_author_repository)
_author_dto_mapper = AuthorDTOMapper()

author_module = {
    "service": _author_service,
    "dto_mapper": _author_dto_mapper,
    "dto_schema": author_dto
}

File: Javascript/backend_nest/src/framework/modules/AuthorModule.ts
import { Module } from '@nestjs/common';
import { AuthorController } from '../controllers/AuthorController';
import { AuthorService } from 'src/services/AuthorService';
import { AuthorRepository } from 'src/domain/repositories/AuthorRepository';
import { AuthorInMemoryMapper } from 'src/infra/db/inmemory/mappers/AuthorInMemoryMapper';
import { AuthorInMemoryRepository } from 'src/infra/db/inmemory/repositories/AuthorInMemoryRepository';
import { AuthorinMemoryDB } from 'src/infra/db/inmemory/data/AuthorDB';
import { AuthorDTOMapper } from 'src/dto/mapper/AuthorDTOMapper';
import { Mapper } from 'src/domain/kernel/Mapper';
import { Author } from 'src/domain/entities/Author';

@Module({
  imports: [],
  controllers: [AuthorController],
  providers: [
    {
      provide: 'DTOMapper',
      useClass: AuthorDTOMapper, // On injecte le mapper pour les DTOs
    },
    {
      provide: 'DatabaseMapper',
      useClass: AuthorInMemoryMapper, // Le moment où on passe sur autre chose que du inmemory, on peut changer cette ligne pour l'injection
    },
    {
      provide: 'Repository',
      useFactory: (mapper: Mapper<Author, any>) => new AuthorInMemoryRepository(AuthorinMemoryDB, mapper), // Idem ici, on change juste la classe de repository
      inject: ['DatabaseMapper'],
    },
    {
      provide: 'Service',
      useFactory: (authorRepository: AuthorRepository) => new AuthorService(authorRepository), // Ici on pourrait décider d'utiliser un autre service
      inject: ['Repository'],
    },
  ],
})
export class AuthorModule {}
